File: c_console/GuessingGame.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
	srand(time(0));
	int rangeStart = 1;
	int rangeEnd = 10;
	int number = rand() % (rangeEnd - rangeStart) + rangeStart;
	printf("Pick a number between %d and %d\n", rangeStart, rangeEnd);

	int notGuessed = -1;
	int guessCount = 0;
	while (notGuessed)
	{
		char userEntered[255];
		int numberGuessed = 0;
		gets(userEntered);
		numberGuessed = atoi(userEntered);
		if (numberGuessed > 0) {
			guessCount += 1;
			if (numberGuessed == number)
			{
				printf("%d was the number. It took you %d guesses\n", numberGuessed, guessCount);
				notGuessed = 0;
			}
			else
			{
				if (numberGuessed < rangeStart || numberGuessed > rangeEnd)
				{
					printf("That number is not between %d and %d\n", rangeStart, rangeEnd);
					continue;
				}
				if (numberGuessed < number)
				{
					printf("That number is too low\n");
					continue;
				}
				if (numberGuessed > number)
				{
					printf("That number is too high\n");
					continue;
				}
			}
		}
		else {
			printf("Invalid number\n");
		}
	}
}

File: java_console/src/j_console/GuessingGame.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package j_console;

import java.util.Random;
import java.util.Scanner;

/**
 *
 * @author Kevin
 */
public class GuessingGame {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        Random random = new Random();
        Integer rangeStart = 1;
        Integer rangeEnd = 10;
        Integer number = random.nextInt(rangeEnd - rangeStart) + rangeStart;
        System.out.printf("Pick a number between %d and %d.\n", rangeStart, rangeEnd);

        Boolean notGuessed = true;
        Integer guessCount = 0;
        while (notGuessed) {
            String userEntered = input.nextLine();
            Integer numberGuessed = 0;
            
            try {
                numberGuessed = Integer.parseInt(userEntered);
            } catch (NumberFormatException e) {
                System.out.println("Invalid number. Try again.\n");
            }

            guessCount += 1;
            if (numberGuessed == number) {
                System.out.printf("%d was the number. It took you %d guesses.\n", numberGuessed, guessCount);
                notGuessed = false;
            } else {
                if (numberGuessed < rangeStart || numberGuessed > rangeEnd) {
                    System.out.printf("That number is not between %d and %d.\n", rangeStart, rangeEnd);
                    continue;
                }
                if (numberGuessed < number) {
                    System.out.printf("That number is too low.\n");
                    continue;
                }
                if (numberGuessed > number) {
                    System.out.printf("That number is too high.\n");
                    continue;
                }
            }

        }
    }

}
